File: backend/src/MicroclimateIotSystem.Domain/Entities/Device.cs
using MicroclimateIotSystem.Domain.Abstractions;

namespace MicroclimateIotSystem.Domain.Entities;

public class Device : BaseEntity
{
    public string HardwareId { get; set; }
    public string Name { get; set; }
    public bool IsActive { get; set; }
    public int TelemetryIntervalSeconds { get; set; }
    public int? RoomId { get; set; }
    
    public Room? Room { get; set; }
}

File: frontend/src/app/models/device.ts
export interface Device {
  id: number;
  name: string;
  hardwareId: string;
  isActive: boolean;
  telemetryIntervalSeconds: number;
  roomId: number | null;
  roomName: string | null;
}

export interface CreateDeviceRequest {
  name: string;
  hardwareId: string;
  roomId: number | null;
}

export interface UpdateDeviceRequest {
  id: number;
  name: string;
  hardwareId: string;
  isActive: boolean;
  roomId: number | null;
}

export interface DeviceConfigRequest {
  telemetryIntervalSeconds: number;
}